### PR TITLE
chore: update dockerfile

### PR DESCRIPTION
**Pull request by jajera**
update requirements

### CODE REVIEW

File: el8/Dockerfile [hunk 1]
```diff
@@ -5,7 +5,27 @@ COPY . .
 
 # Installing tools needed for rpmbuild ,
 RUN dnf module enable -y nodejs:16
-RUN dnf install -y rpm-build rpmdevtools rpm-sign rpmlint git dnf-plugins-core nodejs npm make automake gcc gcc-c++ kernel-devel python3 glibc
+
+# install base and app requirements
+RUN dnf install -y boost automake boost-filesystem boost-iostreams boost-program-options boost-regex boost-signals boost-system boost-thread cairo cairo-devel dnf-plugins-core epel-release epel-next-release fontconfig fontconfig-devel freetype freetype-devel gcc gcc-c++ git glibc kernel-devel libgfortran libxml2 libxml2-devel make ncurses ncurses-devel nodejs npm openssl pango pango-devel perl-devel python3 rpm-build rpmdevtools rpm-sign rpmlint shadow-utils subscription-manager systemd
+
+# enable powertools
+RUN yum config-manager --set-enabled powertools
```
Comment (Mossman1215) [inline el8/Dockerfile:13]: put this in when the module enable nodejs is called in the same line using the && operator
this saves on calls to yum servers and layers better

**Comment by Mossman1215** [comment]
> update requirements

I need more details in the git comment history
specifically addressing why you made the change.
or at minimum linking the ticket.

File: el8/Dockerfile (hunk 1 above)
Comment (jajera) [reply el8/Dockerfile]: update the code

File: el8/Dockerfile [hunk 2]
```diff
@@ -5,10 +5,31 @@ COPY . .
...
+RUN dnf install -y boost automake boost-filesystem boost-iostreams \
+  boost-program-options boost-regex boost-signals boost-system \
+  boost-thread cairo cairo-devel dnf-plugins-core epel-release \
+  epel-next-release fontconfig fontconfig-devel freetype \
+  freetype-devel gcc gcc-c++ git glibc kernel-devel libgfortran \
+  libxml2 libxml2-devel make ncurses ncurses-devel nodejs npm \
+  openssl pango pango-devel perl-devel python3 rpm-build \
+  rpmdevtools rpm-sign rpmlint shadow-utils subscription-manager systemd \
+  # enable powertools
+  && yum config-manager --set-enabled powertools
```
Comment (Mossman1215) [inline el8/Dockerfile:19]: sorry I was meaning line 7 not this one

Comment (Mossman1215) [reply el8/Dockerfile:19]: i think it's going to need to be dnf  rather than yum

**Comment by jajera** [comment]
el7

1. This is the original + those that came from the package requirements.
``` bash
# install base and app requirements
RUN dnf install -y boost automake boost-filesystem boost-iostreams \
  boost-program-options boost-regex boost-signals boost-system \
  boost-thread cairo cairo-devel dnf-plugins-core epel-release \
  epel-next-release fontconfig fontconfig-devel freetype \
  freetype-devel gcc gcc-c++ git glibc kernel-devel libgfortran \
  libxml2 libxml2-devel make ncurses ncurses-devel nodejs npm \
  openssl pango pango-devel perl-devel python3 rpm-build \
  rpmdevtools rpm-sign rpmlint shadow-utils subscription-manager systemd \
```

2. these packages relies on epel and powertools repositories, they have to install after. few of the requirements are python module that has to handled via pip.
``` bash
# trigger an update after enabling epel-release and powertools
RUN dnf update -y \
  # install after epel is available
  && dnf install -y libdbi libdbi-devel octave qt5-qtwebengine \
  # python modules
  && pip3 install --user numpy python-dateutil twisted boto3 \
```

3. need more testing to integrate these packages
``` bash
  # python module having issue
  # pip3 install --user m2crypto
  # in-house
  # seiscomp seiscomp-toast seiscomp3-aio
```

File: el7/Dockerfile [hunk 3]
```diff
@@ -6,10 +6,25 @@ COPY . .
...
+  cairo cairo-devel createrepo epel-release fontconfig fontconfig-devel \
+  freetype freetype-devel gcc gcc-c++ git libdbi libdbi-devel libgfortran \
+  libxml2 libxml2-devel m2crypto make ncurses ncurses-devel numpy openssl \
+  pango pango-devel perl-devel python3 rpm-build rpmdevtools rpmlint rpm-sign \
+  shadow-utils systemd tar yum-utils
+
+# trigger an update after enabling epel-release
+RUN yum update -y \
+  # install after epel is available
+  && yum install -y octave qtwebkit \
```
Comment (Mossman1215) [inline el7/Dockerfile:21]: can you put the yum installs into one block?

**Comment by jajera** [comment]
el8
1. This is the original + those that came from the package requirements.
``` bash
# install base and app requirements
RUN dnf install -y boost automake boost-filesystem boost-iostreams \
  boost-program-options boost-regex boost-signals boost-system \
  boost-thread cairo cairo-devel dnf-plugins-core epel-release \
  epel-next-release fontconfig fontconfig-devel freetype \
  freetype-devel gcc gcc-c++ git glibc kernel-devel libgfortran \
  libxml2 libxml2-devel make ncurses ncurses-devel nodejs npm \
  openssl pango pango-devel perl-devel python3 rpm-build \
  rpmdevtools rpm-sign rpmlint shadow-utils subscription-manager systemd \
  # enable powertools
  && yum config-manager --set-enabled powertools
```

2. these packages relies on epel and powertools repositories, they have to install after. few of the requirements are python module that has to handled via pip.
``` bash
# trigger an update after enabling epel-release and powertools
RUN dnf update -y \
  # install after epel is available
  && dnf install -y libdbi libdbi-devel octave qt5-qtwebengine \
  # python modules
  && pip3 install --user numpy python-dateutil twisted boto3 \
  # python module having issue
  # pip3 install --user m2crypto
  # in-house
  # seiscomp seiscomp-toast seiscomp3-aio
```

3. need more testing to integrate these packages
``` bash
  # python module having issue
  # pip3 install --user m2crypto
  # in-house
  # seiscomp seiscomp-toast seiscomp3-aio
```

File: el8/Dockerfile (hunk 2 above)
Comment (jajera) [reply el8/Dockerfile:19]: updating the code

File: el7/Dockerfile (hunk 3 above)
Comment (jajera) [reply el7/Dockerfile:21]: based on testing, it fails to install if its on the same yum install with epel-release. getting run after epel-release install and yum update gets it installed successfully.

File: el8/Dockerfile [hunk 4]
```diff
@@ -3,12 +3,32 @@ FROM ghcr.io/geonet/base-images/centos:stream8
...
+  'dnf-command(config-manager)' \
+  && dnf config-manager --set-enabled powertools
 
-RUN npm install \
-    && npm run-script build
+# Update system
+RUN dnf update -y
+
+# Install dependencies
+RUN dnf install -y boost automake boost-filesystem boost-iostreams \
```
Comment (Mossman1215) [inline el8/Dockerfile:16]: last request
can you separate the node and pip install from the dnf command?
and remove any commented packages we've not figured out a process for?

Comment (jajera) [reply el8/Dockerfile:16]: code updated.

**Review by Mossman1215 — APPROVED** [review]
Righto will give this a try.

**Pull request merged**